components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt

```diff
@@ -389,6 +390,36 @@ class TabsUseCases(
         }
     }
 
+    /**
+     * Use case for duplicating a tab.
+     */
+    class DuplicateTabUseCase(
+        private val sessionManager: SessionManager
+    ) {
+        /**
+         * Creates a duplicate of [tab] (including history) and selects it if [selectNewTab] is true.
+         */
+        operator fun invoke(
+            tab: TabSessionState,
+            selectNewTab: Boolean = true
+        ) {
+            val duplicate = Session(
+                initialUrl = tab.content.url,
+                private = tab.content.private,
+                contextId = tab.contextId
```

Comment:
Would it be valuable to have a separate `SessionState.Source` set for duplication?